guides/source/form_helpers.md

```diff
@@ -64,184 +64,229 @@ To create this form you will use `form_with` and the form builder object it yiel
 
 ### Other Helpers of Interest
 
-Other form controls worth mentioning are text areas, hidden fields, password fields, number fields, date and time fields, and many more:
+There are many other form controls for text, email, password, date, and time. Below examples show some more helpers and their generated HTML.
+
+Date and time related helpers:
 
 ```erb
-<%= form.text_area :message, size: "70x5" %>
-<%= form.hidden_field :parent_id, value: "foo" %>
-<%= form.password_field :password %>
-<%= form.number_field :price, in: 1.0..20.0, step: 0.5 %>
-<%= form.range_field :discount, in: 1..100 %>
 <%= form.date_field :born_on %>
 <%= form.time_field :started_at %>
 <%= form.datetime_local_field :graduation_day %>
 <%= form.month_field :birthday_month %>
 <%= form.week_field :birthday_week %>
-<%= form.search_field :name %>
+```
+
+Output:
+
+```html
+<input type="date" name="born_on" id="born_on">
+<input type="time" name="started_at" id="started_at">
+<input type="datetime-local" name="graduation_day" id="graduation_day">
+<input type="month" name="birthday_month" id="birthday_month">
+<input type="week" name="birthday_week" id="birthday_week">
+```
+
+Helpers with special formatting:
+
+```erb
+<%= form.password_field :password %>
 <%= form.email_field :address %>
 <%= form.telephone_field :phone %>
 <%= form.url_field :homepage %>
+```
+
+Output:
+
+```html
+<input type="password" name="password" id="password">
+<input type="email" name="address" id="address">
+<input type="tel" name="phone" id="phone">
+<input type="url" name="homepage" id="homepage">
+```
+
+Other common helpers:
+
+```erb
+<%= form.text_area :message, size: "70x5" %>
+<%= form.hidden_field :parent_id, value: "foo" %>
+<%= form.number_field :price, in: 1.0..20.0, step: 0.5 %>
+<%= form.range_field :discount, in: 1..100 %>
+<%= form.search_field :name %>
 <%= form.color_field :favorite_color %>
 ```
 
 Output:
 
 ```html
 <textarea name="message" id="message" cols="70" rows="5"></textarea>
-<input type="hidden" name="parent_id" id="parent_id" value="foo" />
-<input type="password" name="password" id="password" />
-<input type="number" name="price" id="price" step="0.5" min="1.0" max="20.0" />
-<input type="range" name="discount" id="discount" min="1" max="100" />
-<input type="date" name="born_on" id="born_on" />
-<input type="time" name="started_at" id="started_at" />
-<input type="datetime-local" name="graduation_day" id="graduation_day" />
-<input type="month" name="birthday_month" id="birthday_month" />
-<input type="week" name="birthday_week" id="birthday_week" />
-<input type="search" name="name" id="name" />
-<input type="email" name="address" id="address" />
-<input type="tel" name="phone" id="phone" />
-<input type="url" name="homepage" id="homepage" />
-<input type="color" name="favorite_color" id="favorite_color" value="#000000" />
+<input value="foo" autocomplete="off" type="hidden" name="parent_id" id="parent_id">
```

Comment:
It would be great to set `type` as first attribute to be consistent with other examples.

Reply:
I agree it would look nice, but this currently is the actual output of these helpers.
Maybe we should change the implementation to render these consistently instead.

Reply:
Yeah, I copy/pasted output from an app.